.github/workflows/build.yml

```diff
@@ -2,31 +2,32 @@ name: build
 on:
   pull_request:
   push:
-      branches:
-        - main
+    branches:
+      - main
 jobs:
   build:
     runs-on: ubuntu-latest
-    continue-on-error: ${{ matrix.experimental }}
     strategy:
       fail-fast: false
       matrix:
-        ruby-version: [2.7.8]
-        experimental: [false]
-        include:
-          - ruby-version: 3.0
-            experimental: true
-          - ruby-version: 3.1
-            experimental: true
-          - ruby-version: 3.2
-            experimental: true
+        ruby-version: [3.0, 3.1, 3.2]
     steps:
       - uses: actions/checkout@v4
         with:
-          show-progress: 'false'
+          show-progress: "false"
+      - name: get bundler version
+        run: echo "BUNDLER=$(cat .bundler-version)" >> $GITHUB_ENV
       - uses: ruby/setup-ruby@v1
         with:
           ruby-version: ${{ matrix.ruby-version }}
-          bundler-cache: true # runs `bundle install` and caches installed gems automatically
+          bundler: ${{ env.BUNDLER }}
+          bundler-cache: false
```

Comment:
If I'm reading the source code of `ruby/setup-ruby` right, I believe the ruby-version is part of the cache key, so it should be okay to continue using it. Unless there's a separate concern I'm not thinking about?

Reply:
since we are installing in a tmp folder to the side, and it's so fast, there's no real value in caching here, better to be clean to be sure. we also want to control the bundle install timing our selves due to appraisal usage

Reply:
Fixing in https://github.com/guideline-tech/subroutine/pull/67 as the newer bundler fixed this issue